### PR TITLE
AWS CodeBuild support

### LINKED ISSUE #37: Add AWS CodeBuild support
See https://aws.amazon.com/codebuild/

### PR DESCRIPTION
**Pull request by radimvaculik**
Closes #37.

### CODE REVIEW

File: src/Ci/AwsCodeBuild.php [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    }
+
+    public function getCiName(): string
+    {
+        return CiDetector::CI_AWS_CODEBUILD;
+    }
+
+    public function isPullRequest(): TrinaryLogic
+    {
+        return TrinaryLogic::createFromBoolean(mb_substr($this->env->getString('CODEBUILD_WEBHOOK_EVENT'), 0, mb_strlen('PULL_REQUEST')) === 'PULL_REQUEST');
```
Comment (OndraM) [inline src/Ci/AwsCodeBuild.php]:
```suggestion
        return TrinaryLogic::createFromBoolean(
            mb_strpos($this->env->getString('CODEBUILD_WEBHOOK_EVENT'), 'PULL_REQUEST') === 0
        );
```

File: src/Ci/AwsCodeBuild.php [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    }
+
+    public function getBuildNumber(): string
+    {
+        return $this->env->getString('CODEBUILD_BUILD_ID');
```
Comment (OndraM) [inline src/Ci/AwsCodeBuild.php]: Would `CODEBUILD_BUILD_NUMBER` be more suitable? 

`CODEBUILD_BUILD_ID` is more like "unique build identifier".

File: tests/phpt/Ci/CiDetector_AwsCodeBuildPullRequest.phpt [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+require __DIR__ . '/../../../vendor/autoload.php';
+
+$ci = (new OndraM\CiDetector\CiDetector())->detect();
+echo "Is pull request:\n";
+var_dump($ci->isPullRequest()->describe());
+
+--EXPECT--
+Is pull request:
+string(3) "Yes"
```
Comment (OndraM) [inline tests/phpt/Ci/CiDetector_AwsCodeBuildPullRequest.phpt:124]: I'd add test for the branch name, something like:

```suggestion
echo "Git branch:\n";
var_dump($ci->getGitBranch());

--EXPECT--
Is pull request:
string(3) "Yes"
Git branch:
string(8) "test/pr-branch"
```

**Review by OndraM — CHANGES_REQUESTED** [review]
:+1: Thanks a lot @radimvaculik! And sorry for delay with the review.

File: src/Ci/AwsCodeBuild.php (hunk 2 above)
Comment (radimvaculik) [reply src/Ci/AwsCodeBuild.php]: There is no another identifier 🤷

Comment (OndraM) [reply src/Ci/AwsCodeBuild.php]: What do you mean? :)

CODEBUILD_BUILD_NUMBER has value like "559" etc. which is more "build number" than  values like "project-api-pr:cf59169d-ddc7-4292-a5ab-06e6f5b8c8c4" which are in CODEBUILD_BUILD_ID. 

I see build number more like human readable number of the build, not unique identifier.

Using the number is also more consistent with other CI - eg. in Travis it is used "TRAVIS_JOB_NUMBER" (with values like "25.2") and not "TRAVIS_BUILD_ID" (with values like "148395135").

Comment (radimvaculik) [reply src/Ci/AwsCodeBuild.php]: 🤦 You're right 👍

**Comment by radimvaculik** [comment]
@OndraM I fixed all your requests. Thanks for review 👍

**Comment by OndraM** [comment]
Thanks for the contribution! :100:

**Pull request merged**